File: src/buses/dto/update-kilometraje.dto.ts
import { IsNotEmpty, IsNumber, Min } from 'class-validator';

export class UpdateKilometrajeDto {
  @IsNumber()
  @IsNotEmpty({ message: 'El kilometraje es requerido' })
  @Min(0, { message: 'El kilometraje no puede ser negativo' })
  kmB: number;
}

File: src/main/java/com/tallermecanico/dto/request/UpdateKilometrajeDto.java
package com.tallermecanico.dto.request;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

public class UpdateKilometrajeDto {

    @NotNull(message = "El kilometraje es requerido")
    @Min(value = 0, message = "El kilometraje no puede ser negativo")
    private Integer kmB;

    // Constructors
    public UpdateKilometrajeDto() {
    }

    public UpdateKilometrajeDto(Integer kmB) {
        this.kmB = kmB;
    }

    // Getters and Setters
    public Integer getKmB() {
        return kmB;
    }

    public void setKmB(Integer kmB) {
        this.kmB = kmB;
    }
}
